code/turf/floors.dm

```diff
@@ -1048,6 +1048,16 @@ DEFINE_FLOORS(minitiles/black,
 	name = "boxing mat"
 	icon_state = "boxing"
 
+	Entered(atom/movable/M, atom/OldLoc)
+		..()
+		if (!istype(OldLoc, /turf/simulated/floor/specialroom/gym))
```

Comment:
What is the purpose of adding the wrestling status on entering the ring? Why not just add it when the bell is rung?